tests/models/test_bootstrap_election_model.py

```diff
@@ -579,6 +579,72 @@ def test_is_top_level_aggregate(bootstrap_election_model):
     assert not bootstrap_election_model._is_top_level_aggregate([])
 
 
+def test_format_called_contests_dictionary(bootstrap_election_model, rng):
+    n_contests = 10
+    bootstrap_election_model.n_contests = n_contests
```

Comment:
I've noticed this in a bunch of places.  Why do we have to set this here instead of getting it from `aggregate_indicator.shape[1]` like we do in the `BootstrapElectionModel` class? 🤔

Reply:
because the aggregate indicator does not exist in the unit test, since we are never forcing the model to run the bootstrap.

Reply:
Ok, I see, so for testing purposes we want to bypass forcing the model to run the bootstrap so we can test this specific functionality.  I was just curious how common this could be and if we should add something to the `BootstrapElectionModel` class, but for testing, this is good 😄 🎉

Reply:
It is actually pretty common! What do you think we could add there? 👀

Reply:
Oh wow!  Ok, in that case, what about this:

```python
class BootstrapElectionModel:
    def __init__(self, <current arguments>, n_contests=None): # or whatever n_contests default value you think is best
      # other code etc., then
       if aggregate_indicator is None:
          self.n_contests = n_contests
      else:
          self.n_contests = aggregate_indicator.shape[1]
```

I think that should work 🤔

Reply:
Although, in these unit tests `BootstrapElectionModel` is already instantiated so that doesn't help you 😢  You could also add a setter:

```python
# add self.n_contests to def __init__(), then

@n_contests.setter
def n_contests(self, n):
    # you can do some verification on `n` if you wanted to here
    self.n_contests = n
```

But that really doesn't give you anything special above what you have right now, other than people can't get `n_contests` without you adding a getter, although it does allow you to add in some verification if you needed it 🤔 🤔

Reply:
hmm in that case lets leave it for now (but thanks for looking into this!)